Lib/timeit.py

```diff
@@ -257,53 +296,16 @@ def main(args=None, *, _wrap_timer=None):
     is not None, it must be a callable that accepts a timer function
     and returns another timer function (used for unit testing).
     """
-    if args is None:
-        args = sys.argv[1:]
-    import getopt
-    try:
-        opts, args = getopt.getopt(args, "n:u:s:r:pvh",
-                                   ["number=", "setup=", "repeat=",
-                                    "process", "verbose", "unit=", "help"])
-    except getopt.error as err:
-        print(err)
-        print("use -h/--help for command line help")
-        return 2
-
-    timer = default_timer
-    stmt = "\n".join(args) or "pass"
-    number = 0  # auto-determine
-    setup = []
-    repeat = default_repeat
-    verbose = 0
-    time_unit = None
-    units = {"nsec": 1e-9, "usec": 1e-6, "msec": 1e-3, "sec": 1.0}
-    precision = 3
-    for o, a in opts:
-        if o in ("-n", "--number"):
-            number = int(a)
-        if o in ("-s", "--setup"):
-            setup.append(a)
-        if o in ("-u", "--unit"):
-            if a in units:
-                time_unit = a
-            else:
-                print("Unrecognized unit. Please select nsec, usec, msec, or sec.",
-                      file=sys.stderr)
-                return 2
-        if o in ("-r", "--repeat"):
-            repeat = int(a)
-            if repeat <= 0:
-                repeat = 1
-        if o in ("-p", "--process"):
-            timer = time.process_time
-        if o in ("-v", "--verbose"):
-            if verbose:
-                precision += 1
-            verbose += 1
-        if o in ("-h", "--help"):
-            print(__doc__, end="")
-            return 0
-    setup = "\n".join(setup) or "pass"
+    parser = _make_parser()
+    args = parser.parse_args(args)
+
+    setup = "\n".join(args.setup) or "pass"
+    stmt = "\n".join(args.statement) or "pass"
```

Comment:
```suggestion
    statement_parts = [part for part in args.statement if part != '--']
    stmt = "\n".join(statement_parts) or "pass"
```

combined with removing the dummy option:

```
-    # add a dummy option to document '--'
     group.add_argument(
-        "--", dest="_", action="store_const", const=None,
-        help="separate options from statement; "
-             "use when statement starts with -",
-    )
```

brought back the `-- behavior`.

Reply:
No indeed it's an error on my side. Thanks for catching this.

Reply:
By the way, `--` should only be removed if this is the first statement. If someone uses `--` inside the statements, we should still have a SyntaxError (current behavior).